5.5/root/usr/bin/run-mysqld-slave

```diff
@@ -31,31 +25,34 @@ envsubst < ${CONTAINER_SCRIPTS_PATH}/my-paas.cnf.template > /etc/my.cnf.d/paas.c
 envsubst < ${CONTAINER_SCRIPTS_PATH}/my-slave.cnf.template > /etc/my.cnf.d/slave.cnf
 envsubst < ${CONTAINER_SCRIPTS_PATH}/my-tuning.cnf.template > /etc/my.cnf.d/tuning.cnf
 
-# Initialize MySQL database and wait for the MySQL master to accept
-# connections.
-initialize_database "$@"
-wait_for_mysql_master
+if [ ! -e "${MYSQL_DATADIR}/mysql" ]; then
```

Comment:
Sorry I misunderstood your sentence at first.

I didn't do extensive testing with loads of data, but just went far enough to verify that it works.


The slave log after restart:

```
--> 11:11:12     The 'slave' server-id is 1657242886
---> 11:11:12     Processing MySQL configuration files ...
---> 11:11:12     Cleaning up environment variables MYSQL_USER, MYSQL_PASSWORD, MYSQL_DATABASE and MYSQL_ROOT_PASSWORD ...
---> 11:11:12     Running final exec -- Only MySQL server logs after this point
160526 11:11:12 [Note] /opt/rh/mysql55/root/usr/libexec/mysqld (mysqld 5.5.45-log) starting as process 1 ...
160526 11:11:12 [Note] Plugin 'FEDERATED' is disabled.
160526 11:11:12 InnoDB: The InnoDB memory heap is disabled
160526 11:11:12 InnoDB: Mutexes and rw_locks use GCC atomic builtins
160526 11:11:12 InnoDB: Compressed tables use zlib 1.2.7
160526 11:11:12 InnoDB: Using Linux native AIO
160526 11:11:12 InnoDB: Initializing buffer pool, size = 32.0M
160526 11:11:12 InnoDB: Completed initialization of buffer pool
160526 11:11:12 InnoDB: highest supported file format is Barracuda.
InnoDB: Log scan progressed past the checkpoint lsn 1701194
160526 11:11:12  InnoDB: Database was not shut down normally!
InnoDB: Starting crash recovery.
InnoDB: Reading tablespace information from the .ibd files...
InnoDB: Restoring possible half-written data pages from the doublewrite
InnoDB: buffer...
InnoDB: Doing recovery: scanned up to log sequence number 1701204
InnoDB: Last MySQL binlog file position 0 107, file name /var/lib/mysql/data/mysql-bin.000004
160526 11:11:12  InnoDB: Waiting for the background threads to start
160526 11:11:13 InnoDB: 5.5.45 started; log sequence number 1701204
160526 11:11:13 [Note] Recovering after a crash using /var/lib/mysql/data/mysql-bin
160526 11:11:13 [Note] Starting crash recovery...
160526 11:11:13 [Note] Crash recovery finished.
160526 11:11:13 [Note] Server hostname (bind-address): '0.0.0.0'; port: 3306
160526 11:11:13 [Note]   - '0.0.0.0' resolves to '0.0.0.0';
160526 11:11:13 [Note] Server socket created on IP: '0.0.0.0'.
160526 11:11:13 [Warning] 'user' entry 'root@mysql-slave-1-83ipc' ignored in --skip-name-resolve mode.
160526 11:11:13 [Warning] 'user' entry '@mysql-slave-1-83ipc' ignored in --skip-name-resolve mode.
160526 11:11:13 [Warning] 'proxies_priv' entry '@ root@mysql-slave-1-83ipc' ignored in --skip-name-resolve mode.
160526 11:11:13 [Note] Event Scheduler: Loaded 0 events
160526 11:11:13 [Note] /opt/rh/mysql55/root/usr/libexec/mysqld: ready for connections.
Version: '5.5.45-log'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  MySQL Community Server (GPL)
160526 11:11:13 [Note] Slave SQL thread initialized, starting replication in log 'mysql-bin.000004' at position 27399, relay log '/var/lib/mysql/data/mysql-relay-bin.000057' position: 253
160526 11:11:13 [Note] Slave I/O thread: connected to master 'master@mysql-master:3306',replication started in log 'mysql-bin.000004' at position 27399
```

... suggests that it picked up from where it left.

Then, I wrote more data in the master and it was successfully synced to the slave.


In a separate restart cycle, as I write this comment, I decided to write data to master while the slave is down and ensure that it gets synced when the slave comes back up.
Result: items written to master while slave was down were successfully synced after restart.

Log says:

```
---> 15:23:30     The 'slave' server-id is 1657242886
---> 15:23:30     Processing MySQL configuration files ...
---> 15:23:30     Cleaning up environment variables MYSQL_USER, MYSQL_PASSWORD, MYSQL_DATABASE and MYSQL_ROOT_PASSWORD ...
---> 15:23:30     Running final exec -- Only MySQL server logs after this point
160526 15:23:30 [Note] /opt/rh/mysql55/root/usr/libexec/mysqld (mysqld 5.5.45-log) starting as process 1 ...
160526 15:23:30 [Note] Plugin 'FEDERATED' is disabled.
160526 15:23:30 InnoDB: The InnoDB memory heap is disabled
160526 15:23:30 InnoDB: Mutexes and rw_locks use GCC atomic builtins
160526 15:23:30 InnoDB: Compressed tables use zlib 1.2.7
160526 15:23:30 InnoDB: Using Linux native AIO
160526 15:23:30 InnoDB: Initializing buffer pool, size = 32.0M
160526 15:23:30 InnoDB: Completed initialization of buffer pool
160526 15:23:30 InnoDB: highest supported file format is Barracuda.
160526 15:23:30  InnoDB: Waiting for the background threads to start
160526 15:23:31 InnoDB: 5.5.45 started; log sequence number 1701412
160526 15:23:31 [Note] Server hostname (bind-address): '0.0.0.0'; port: 3306
160526 15:23:31 [Note]   - '0.0.0.0' resolves to '0.0.0.0';
160526 15:23:31 [Note] Server socket created on IP: '0.0.0.0'.
160526 15:23:31 [Warning] 'user' entry 'root@mysql-slave-1-83ipc' ignored in --skip-name-resolve mode.
160526 15:23:31 [Warning] 'user' entry '@mysql-slave-1-83ipc' ignored in --skip-name-resolve mode.
160526 15:23:31 [Warning] 'proxies_priv' entry '@ root@mysql-slave-1-83ipc' ignored in --skip-name-resolve mode.
160526 15:23:31 [Note] Slave SQL thread initialized, starting replication in log 'mysql-bin.000004' at position 27629, relay log '/var/lib/mysql/data/mysql-relay-bin.000059' position: 483
160526 15:23:31 [Note] Event Scheduler: Loaded 0 events
160526 15:23:31 [Note] /opt/rh/mysql55/root/usr/libexec/mysqld: ready for connections.
Version: '5.5.45-log'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  MySQL Community Server (GPL)
160526 15:23:31 [Note] Slave I/O thread: connected to master 'master@mysql-master:3306',replication started in log 'mysql-bin.000004' at position 27629
```


Notice that in the last line the replication is started from different positions at the binlog.